nncf/experimental/torch/nas/bootstrapNAS/elasticity/elastic_depth.py

```diff
@@ -0,0 +1,362 @@
+
+    def __init__(self, elasticity_params: Optional[Dict[str, Any]] = None,
+                 ignored_scopes: Optional[List[str]] = None,
+                 target_scopes: Optional[List[str]] = None):
+        super().__init__(ignored_scopes, target_scopes, elasticity_params)
+        self._mode = ElasticDepthMode.from_str(self._elasticity_params.get('mode', ElasticDepthMode.AUTO.value))
+        self._skipped_blocks = None  # type: Optional[BUILDING_BLOCKS]
+        self._skip_dependencies = None  # type: Optional[GROUPED_BLOCK_IDS]
+        self._ordinal_ids = None  # type: Optional[ORDINAL_IDS]
+        self._max_block_size = self._elasticity_params.get('max_block_size', 50)
+        self._min_block_size = self._elasticity_params.get('min_block_size', 6)
+        self._allow_nested_blocks = self._elasticity_params.get('allow_nested_blocks', False)
+        self._allow_linear_combination = self._elasticity_params.get('allow_linear_combination', False)
+
+        if self._mode == ElasticDepthMode.MANUAL:
+            skipped_blocks = self._elasticity_params.get('skipped_blocks', [])  # type: List[List[str,str]]
+            self._skipped_blocks = [BuildingBlock(*b) for b in skipped_blocks]
+            self._skip_dependencies = get_group_of_dependent_blocks(self._skipped_blocks)
+
+    def build(self, target_model: NNCFNetwork) -> ElasticDepthHandler:
+        """
+        Creates modifications to the given NNCFNetwork for introducing elastic depth and creates a handler object that
+        can manipulate this elasticity.
+
+        :param target_model: a target NNCFNetwork for adding modifications
+        :return: a handler object that can manipulate the elastic depth.
+        """
+        tracing_context = target_model.get_tracing_context()
+        tracing_context.elastic_depth = True
+
+        if self._mode == ElasticDepthMode.AUTO:
+            if not self._skipped_blocks and not self._skip_dependencies:
+                self._skipped_blocks, self._ordinal_ids, self._skip_dependencies = \
+                    get_building_blocks(
+                        target_model,
+                        max_block_size=self._max_block_size,
+                        min_block_size=self._min_block_size,
+                        allow_nested_blocks=self._allow_nested_blocks,
+                        allow_linear_combination=self._allow_linear_combination
+                    )
+                str_bs = [str(block) for block in self._skipped_blocks]
+                nncf_logger.info('\n'.join(['\n\"Found building blocks:\": [', ',\n'.join(str_bs), ']']))
+        else:
+            graph = target_model.get_original_graph()
+            ordinal_ids = []
+            for block in self._skipped_blocks:
+                start_node = graph.get_node_by_name(block.start_node_name)
+                end_node = graph.get_node_by_name(block.end_node_name)
+                ordinal_ids.append([start_node.node_id, end_node.node_id])
```

Comment:
@asenina @jpablomch ordinal_ids are not needed for Elastic Depth in manual mode anymore.